src/xb-node.c

```diff
@@ -365,6 +389,70 @@ xb_node_get_attr_as_uint (XbNode *self, const gchar *name)
 	return g_ascii_strtoull (tmp, NULL, 10);
 }
 
+/**
+ * xb_node_attr_iter_init:
+ * @iter: an uninitialized #XbNodeIter
+ * @self: a #XbNode
+ *
+ * Initializes a name/value pair iterator for the node attributes
+ * and associates it with @self.
```

Comment:
Might want to mention whether any `clear()` or `free()` function eventually needs to be called on a `XbNodeIter` initialised this way; or explicitly state that’s not necessary.